operate/client/src/App/Processes/MigrationView/TopPanel/Diagrams/v2/TargetDiagram.tsx

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+import {processInstanceMigrationStore} from 'modules/stores/processInstanceMigration';
+import {Header} from '../Header';
+import {DiagramWrapper} from '../styled';
+import {observer} from 'mobx-react';
+import {TargetProcessField} from '../TargetProcessField';
+import {TargetVersionField} from '../TargetVersionField';
+import {processesStore} from 'modules/stores/processes/processes.migration';
+import {processXmlStore} from 'modules/stores/processXml/processXml.migration.target';
+import {DiagramShell} from 'modules/components/DiagramShell';
+import {Diagram} from 'modules/components/Diagram';
+import {useEffect} from 'react';
+import {diagramOverlaysStore} from 'modules/stores/diagramOverlays';
+import {ModificationBadgeOverlay} from 'App/ProcessInstance/TopPanel/ModificationBadgeOverlay';
+import {useProcessInstanceFilters} from 'modules/hooks/useProcessInstancesFilters';
+import {processInstancesSelectionStore} from 'modules/stores/processInstancesSelection';
+import {useProcessInstancesFlowNodeStates} from 'modules/queries/processInstancesStatistics/useFlowNodeStates';
+
+const OVERLAY_TYPE = 'migrationTargetSummary';
+
+const TargetDiagram: React.FC = observer(() => {
+  const {
+    migrationState: {selectedTargetVersion},
+    selectedTargetProcessId,
+  } = processesStore;
+  const isDiagramLoading = processXmlStore.state.status === 'fetching';
+  const isVersionSelected = selectedTargetVersion !== null;
+  const {isSummaryStep} = processInstanceMigrationStore;
+  const stateOverlays = diagramOverlaysStore.state.overlays.filter(
+    ({type}) => type === OVERLAY_TYPE,
+  );
+
+  useEffect(() => {
+    if (selectedTargetProcessId !== undefined) {
+      processXmlStore.fetchProcessXml(selectedTargetProcessId);
+    } else {
+      processXmlStore.reset();
+    }
+    processInstanceMigrationStore.setTargetProcessDefinitionKey(
+      selectedTargetProcessId ?? null,
+    );
+  }, [selectedTargetProcessId]);
+
+  const processInstanceFilters = useProcessInstanceFilters();
+
+  const {data: flowNodeData} = useProcessInstancesFlowNodeStates(
+    {
+      ...processInstanceFilters,
+      processInstanceKey: {
+        $in: processInstancesSelectionStore.selectedProcessInstanceIds,
+      },
```

Comment:
Same as commented here: https://github.com/camunda/camunda/pull/29486#discussion_r2012942578

```suggestion
      processInstanceKey: getProcessInstanceKey(),
```